code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm

```diff
@@ -2,1474 +2,1217 @@
+	new /obj/effect/temp_visual/behemoth/earth_pillar/creation/destruction(loc)
+	if(xeno_owner)
+		SEND_SIGNAL(src, COMSIG_EARTH_PILLAR_DESTROY) // Surprised Destroy() doesn't send a signal, but hey, we make do.
+		xeno_owner = null
+	if(current_holder)
+		force_drop()
+	return ..()
 
-/obj/effect/temp_visual/behemoth/stomp/east/Initialize(mapload, direction)
+/* to do: need damage icons
+/obj/structure/xeno/earth_pillar/update_icon_state()
 	. = ..()
-	switch(direction)
-		if(NORTH)
-			pixel_x -= 12
-			pixel_y += 1
-		if(SOUTH)
-			pixel_x += 10
-			pixel_y -= 1
-		if(WEST)
-			pixel_x -= 11
-			pixel_y -= 1
-		if(EAST)
-			pixel_x += 18
-			pixel_y -= 1
+	if(obj_integrity <= max_integrity * 0.25)
+		icon_state = "earth_pillar_3"
+		return
+	if(obj_integrity <= max_integrity * 0.5)
+		icon_state = "earth_pillar_2"
+		return
+	if(obj_integrity <= max_integrity * 0.75)
+		icon_state = "earth_pillar_1"
+		return
+*/
+
+// Adds interactions specifically for Behemoths.
+/obj/structure/xeno/earth_pillar/attack_alien(mob/living/carbon/xenomorph/xeno_attacker, damage_amount = xeno_attacker.xeno_caste.melee_damage * xeno_attacker.xeno_melee_damage_modifier, damage_type = xeno_attacker.xeno_caste.melee_damage_type, armor_type = xeno_attacker.xeno_caste.melee_damage_armor, effects = TRUE, armor_penetration = xeno_attacker.xeno_caste.melee_ap, isrightclick = FALSE)
+	if(!isxenobehemoth(xeno_attacker)) // If you're not a Behemoth, then you're unworthy of the rock. Begone, pleb.
+		return
+	if(xeno_attacker.a_intent == INTENT_HELP) // Repairs the rock if it's damaged.
+		if(obj_integrity >= max_integrity)
+			balloon_alert(xeno_attacker, "[name] does not need repairs")
+			return
+		while(do_after(xeno_attacker, EARTH_PILLAR_REPAIR_DELAY, NONE, src, BUSY_ICON_CLOCK))
+			var/repair_amount = max_integrity * EARTH_PILLAR_REPAIR_AMOUNT
+			repair_damage(repair_amount, xeno_attacker)
+			var/datum/personal_statistics/xeno_stats = GLOB.personal_statistics_list[xeno_attacker.ckey]
+			xeno_stats.earth_pillar_repairs += repair_amount
+			playsound(src, SFX_BEHEMOTH_EARTH_PILLAR_HIT, 15, TRUE, 5)
+			if(obj_integrity >= max_integrity)
+				balloon_alert(xeno_attacker, "Fully repaired ([obj_integrity]/[max_integrity])")
+				return
+			balloon_alert(xeno_attacker, "+[repair_amount] ([obj_integrity]/[max_integrity])")
+	if(xeno_attacker.a_intent == INTENT_GRAB) // Grabs the rock.
+		when_grabbed(xeno_attacker)
+		return
+	xeno_attacker.do_attack_animation(src)
+	do_jitter_animation(jitter_loops = 1)
+	playsound(src, 'sound/effects/alien/behemoth/earth_pillar_eating.ogg', 30, TRUE)
+	xeno_attacker.visible_message(span_xenowarning("\The [xeno_attacker] eats away at the [src.name]!"), \
+	span_xenonotice(BEHEMOTH_ROCK_EATING_MESSAGES), null, 5)
+	return TRUE
+
+/// Prepares this object to be thrown.
+/obj/structure/xeno/earth_pillar/proc/pre_throw(datum/source)
+	SIGNAL_HANDLER
+	climbable = FALSE
+	pixel_y = initial(pixel_y)
```

Comment:
pixel x and y determine the actual turf you are on for various code and visual calculations, should this be using z and w?